ixmp4/data/api/optimization/column.py

```diff
@@ -0,0 +1,60 @@
+from datetime import datetime
+from typing import ClassVar
+
+from ixmp4.data.abstract import optimization as abstract
+
+from .. import base
+from ..docs import Docs, DocsRepository
+from .indexset import IndexSet
+
+
+class Column(base.BaseModel):
+    NotFound: ClassVar = abstract.Column.NotFound
+    NotUnique: ClassVar = abstract.Column.NotUnique
+    DeletionPrevented: ClassVar = abstract.Column.DeletionPrevented
+
+    id: int
+    name: str
+    dtype: str
+    table__id: int
+    indexset: IndexSet
+    constrained_to_indexset: int
+    unique: bool
+
+    created_at: datetime | None
+    created_by: str | None
+
+
+class ColumnDocsRepository(DocsRepository):
+    model_class = Docs
+    prefix = "docs/optimization/columns/"
+
+
+class ColumnRepository(
+    base.Creator[Column],
+    abstract.ColumnRepository,
+):
+    model_class = Column
+    prefix = "optimization/columns/"
+
+    def __init__(self, client, *args, **kwargs) -> None:
+        super().__init__(client, *args, **kwargs)
+        self.docs = ColumnDocsRepository(self.client)
+
+    # TODO: This is not currently in use, only here because the abstract class has this
```

Comment:
Just to clarify, not in use because the rest api side is not done or not in use because it is only used internally?
In the second case its probably better to omit ColumnRepository from the abstract interface entirely.

Reply:
Not in use because `Table`s only interact with `Column`s on the DB layer level, so I figured there's no need to add an `add_column()` function to `api.Table`, which would only call `api.Column.create()`, which in turn calls `db.Column.create()`. So I guess its kind of all of the above, but I'll just move the docstring to the function in the DB layer and remove the `ColumnRepository` from the abstract layer, as you suggested.